pmd-apex/src/test/java/net/sourceforge/pmd/lang/apex/DefaultRulesetTest.java

```diff
@@ -41,10 +44,11 @@ public void cleanup() {
 
     @Test
     public void loadQuickstartRuleset() throws Exception {
+        List<String> logRecords = new ArrayList<>();
```

Comment:
I'm not sure how useful this test is, but I adapted it to pass after deprecating the 3 rules.

Reply:
I think, the goal of the test is, to verify, when using the quickstart ruleset, that you don't get deprecation warnings....

That means: `run.sh pmd -R rulesets/apex/quickstart.xml ....` shouldn't warn you about deprecations...

I'll have a look, what exactly the problem is here.... (maybe we use a wrong rule set factory here).

Reply:
Ok, it turns out, that we should *not* have deprecated rules in quickstart.xml. I'll remove these 3 old rules. That's ok for this ruleset, since the new rule replaces the 3 old rules.

Referencing the rules via quickstart.xml is not officially supported, e.g. if someone uses the following reference in a custom ruleset, that will break:

```
<rule ref="rulesets/apex/quickstart.xml/AvoidSoqlInLoops" />
```

correct would have been:
```
<rule ref="category/apex/performance.xml/AvoidSoqlInLoops" />
```
This continues to work, but will now issue a deprecation warning.

So, I consider this ok.